test/unittests/util/detect_test.py

```diff
@@ -56,3 +59,16 @@ def test_detect_clang_gcc_toolchain(self, _):
             with environment_update({"CC": "clang-9 --gcc-toolchain=/usr/lib/gcc/x86_64-linux-gnu/9"}):
                 detect_defaults_settings()
                 self.assertIn("CC and CXX: clang-9 --gcc-toolchain", output)
+
+
+@pytest.mark.parametrize("version_return,expected_version", [
+    ["cc.exe (Rev3, Built by MSYS2 project) 14.1.0", "14.1.0"],
+    ["g++ (Conan-Build-gcc--binutils-2.42) 14.1.0", "14.1.0"],
+    ["clang version 18.1.0rc (https://github.com/llvm/llvm-project.git 461274b81d8641eab64d494accddc81d7db8a09e)", "18.1.0"],
```

Comment:
These come from Compiler Explorer gcc's 14.1 and clang's 18.1.0

Reply:
Wow, very good testing inputs, crazy how challenging they can become to parse correctly